nupic/frameworks/opf/model.py

```diff
@@ -126,6 +135,34 @@ def _getLogger(self):
     @returns (Logger) A Logger object, it should not be None.
     """
 
+
+  def _addGlobalModel(self):
+    """ add a model to the global storage, used ie in Metrics
+        Models are stored in a global variable 'globalModelsStorage' 
+        and can be later accessed by name.
+        @param newModel - instance of Model to add
+    """
+    global globalModelsStorage
+
+    newModel = self
+    logger = logging.getLogger() # self._getLogger() not working because of subclass init
+
+    # generate unique name
+    if newModel._name is None:
+      usedNames = globalModelsStorage.keys()
+      id = random.randint(0,1000)
```

Comment:
`id` is builtin (https://docs.python.org/2/library/functions.html#id).  For that matter, you could just assign name to some function of `id()`, which will be unique anyway.  For example `"model-%d" % id(newModel)`